src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -2818,6 +2822,15 @@ MavlinkReceiver::Run()
 			updateParams();
 		}
 
+		// reset cruising speed on mode changes
+		if (_vehicle_status_sub.update(&_vehicle_status)) {
+			if (_last_nav_state != _vehicle_status.nav_state) {
+				reset_cruising_speed();
+				_last_nav_state = _vehicle_status.nav_state;
+			}
+		}
```

Comment:
I think this makes sense if you are exiting offboard modes, but would be quite confusing if you are going through different flight modes within normal flight modes (e.g. stabilized/position)

IMHO, I would rather leave this and make it more transparent. Offboard application on the companion computer should be responsible on ensuring that this is set to a safe value.

Reply:
@Jaeyoung-Lim Is this comment still valid from your side? It really only affects offboard mode, no?